### ISSUE TITLE
Move platform specific window implementations to components

### ISSUE THREAD

**Issue by tomadamatkinson** [KhronosGroup/Vulkan-Samples#699]
Follows on from #698.

Windows make up a large part of the platform folder yet are only needed on specific platforms. They are relatively self contained and because of this should be relatively easy to move.

**Comment by marty-johnson59** [comment]
@tomadamatkinson, should we close this now?

**Comment by SaschaWillems** [comment]
I don't think this is necessary (anymore). The platform part of the framework is stable.
